### ISSUE TITLE
Add more context for the use of directory targets

### ISSUE THREAD

**Issue by hcorson-dosch-usgs** [USGS-R/ds-pipelines-targets-2-course#19]
I chatted with Lindsay and went back and looked at the description of directory targets in issue 3 of the pipelines-II training, and I think we need to add a bit more context to when directory targets are used.

One new employee taking pipelines-II used a directory target in ds-pipelines-II in order to track all of the site data files that would need to be combined into a dataframe, and another new employee mentioned considering it. That set up would look something like this:

```r
  tar_target(
    p1_site_data_01427207_csv,
    {
      site_id = site_nums[1]
      out_file = file.path('1_fetch/out/nwis_data', paste0('nwis_', site_id, '_data.csv'))
      download_nwis_site_data(site_num = site_id,
                              out_file = out_file,
                              parameterCd = parameter_code,
                              startDate = start_date,
                              endDate = end_date)
    },
    format = 'file'
  ),
  tar_target(
    p1_site_data_01432160_csv,
    {
      site_id = site_nums[2]
      out_file = file.path('1_fetch/out/nwis_data', paste0('nwis_', site_id, '_data.csv'))
      download_nwis_site_data(site_num = site_id,
                              out_file = out_file,
                              parameterCd = parameter_code,
                              startDate = start_date,
                              endDate = end_date)
    },
    format = 'file'
  ),
  tar_target(
    p1_nwis_out_dir,
    '1_fetch/out/nwis_data',
    format = 'file'
  ),
  tar_target(
    p1_site_data_csv,
    combine_nwis_data(out_file = '1_fetch/out/site_data.csv',
                      in_dir = p1_nwis_out_dir),
    format = "file"
  ),
```

I've been providing this feedback:
> Directory targets are generally used when the directory already exists and files are already present in that directory _before_ the pipeline is run (e.g., manually added files, the # of which or contents of which might change _between_ pipeline runs, but not _during_ pipeline runs). In the case when file targets are dynamically generated _by the pipeline_ while it is running, it is better to have downstream targets depend explicitly on those file targets, so that all dependencies are clearly tracked. If a directory target is used when files in that directory are being dynamically generated by the pipeline, `targets` does not always catch changes to those files that would affect downstream targets. You'll learn in pipelines-III that `targets` has a better way to track large numbers of very similar targets.

And when I tested it out myself b/c I was curious, `targets` did sometimes catch changes to the datafiles that affected the combined dataframe target, but not always:

> With a file directory set up, because the file directory target is not linked to any preceding targets (like the `site_data_########_csv`  targets), the pipeline will build it at any point, which could be before any site data targets are built. My guess was that in that case, the file directory target would not be rebuilt even if the # or contents of files within it change. But I was curious about how that directory target would behave, so I tested it myself in my pipelines-II code. Right off, I did have to create the directory myself, since the pipeline would merely look for it, but not create it, so that's one immediate issue with the set up. Once the empty directory was there, the pipeline would run, and build out of order. Although interestingly, the `site_data` target was always built after each of the individual `site_data_########` targets, despite nothing explicitly marking it as downstream of those targets, as it was only linked to the directory target. I ran the directory target through a number of tests, changing what data was being placed in the directory and changing the contents of those files, and although it surprisingly _did_ sometimes catch changes to those files, it was inconsistent, and unpredictably so. Sometimes it would mark the directory target and the combined dataframe as OK even when the contents of the data files had changed. So in the end, I don't think that a directory target is a reliable way to track targets generated dynamically by a pipeline. Instead it's best to pass those `site_data_########_csv` targets directly as arguments to the command to build the downstream combined dataframe target.

**Comment by jpadilla-usgs** [comment]
https://github.com/DOI-USGS/ds-pipelines-targets-2-course/commit/c880349f12354995bb07751befb14efdc1e0b876

**Comment by jpadilla-usgs** [comment]
closed in #29
